tao_compiler/mlir/disc/IR/hlo_disc_ops.td

```diff
@@ -61,6 +61,10 @@ def HLO_TensorOrIndexTensor : AnyTypeOf<[HLO_Tensor, IndexTensor]>;
 
 def HLO_I32Tensor : TensorOf<[SignlessIntOfWidths<[32]>]>;
 
+def HLO_I64Tensor : TensorOf<[SignlessIntOfWidths<[64]>]>;
+
+def HLO_IndexTensor : AnyTypeOf<[HLO_I32Tensor, HLO_I64Tensor]>;
```

Comment:
`index` type has special meaning in MLIR, maybe use `I32Or64Tensor`?

Reply:
fixed